src/BaGetter.Database.Sqlite/SqliteContext.cs

```diff
@@ -45,5 +50,49 @@ protected override void OnModelCreating(ModelBuilder builder)
                 .Property(f => f.Moniker)
                 .HasColumnType("TEXT COLLATE NOCASE");
         }
+
+        public override async Task RunMigrationsAsync(CancellationToken cancellationToken)
+        {
+            if (Database.GetDbConnection() is SqliteConnection connection)
+            {
+                /* Create the folder of the Sqlite blob if it does not exist. */
+                CreateSqliteDataSourceDirectory(connection);
+            }
+
+            await base.RunMigrationsAsync(cancellationToken);
+        }
+
+        /// <summary>
+        /// Creates directories specified in the Database::ConnectionString config for the Sqlite database file.
+        /// Respects the difference of root and relative paths.
+        /// </summary>
+        /// <param name="connection">Instance of the <see cref="SqliteConnection"/>.</param>
+        private static void CreateSqliteDataSourceDirectory(SqliteConnection connection)
+        {
+            var configDataSource = connection.DataSource;
+            var pathToCreate = Path.IsPathRooted(configDataSource)
+                // root path
+                ? Path.GetDirectoryName(configDataSource)
+                // relative path
+                : CreateFullDataSourcePath();
+
+            if (pathToCreate is not null)
+            {
+                Directory.CreateDirectory(pathToCreate);
+            }
+
+            return;
+
+            string CreateFullDataSourcePath()
+            {
+                var exeDir = Directory.GetCurrentDirectory();
```

Comment:
I'm not sure `exeDir` is the best description here as this will be the directory it's executed from, not the directory the executable is in.

Reply:
Removed the unecessary function CreateFullDataSourcePath.